components/script/timers.rs

```diff
@@ -231,8 +349,67 @@ impl OneshotTimers {
             if !global.can_continue_running() {
                 return;
             }
-            let callback = timer.callback;
-            callback.invoke(global, &self.js_timers, can_gc);
+            match &timer.callback {
+                OneshotTimerCallback::RunStepsAfterTimeout { ordering_id, .. } => {
+                    // Step 4.2 Wait until any invocations of this algorithm that had the same global and orderingIdentifier,
+                    // that started before this one, and whose milliseconds is less than or equal to this one's, have completed.
+                    let head_handle_opt = {
+                        let queues_ref = self.runsteps_queues.borrow();
+                        queues_ref
+                            .get(ordering_id)
+                            .and_then(|v| v.first().map(|t| t.2))
+                    };
+                    let is_head = head_handle_opt.is_none_or(|head| head == timer.handle);
+
+                    if !is_head {
+                        let rein = OneshotTimer {
```

Comment:
Not sure about this re queuing, but this would go away if we revisit timers as I mentioned in my other comments.